.github/workflows/smithy-rs-sync.yml

```diff
@@ -23,6 +23,12 @@ jobs:
         token: ${{ secrets.AWS_SDK_RUST_CI_PAT }}
         ref: ${{ env.sync_to_branch }}
         path: aws-sdk-rust
+        fetch-depth: 0
```

Comment:
We should consider commenting on why we need this because it's not going to be apparent to most people. I'm thinking something like `We need to fetch the full history because we want to run a rebase from next to main. Without this, the rebase would fail with a merge conflict`